network/server.go

```diff
@@ -370,6 +374,7 @@ func (s *Server) addPeer(id peer.ID) {
 	s.peers[id] = p
 
 	s.emitEvent(id, PeerConnected)
+	s.metrics.Peers.Set(float64(len(s.peers)))
```

Comment:
@dankostiuk I would suggest to acquire peersLock before reading the length. @zivkovicmilos Whats your take ?

Reply:
You're right 💯 

@dankostiuk please call s.numPeers() since `s.peers` requires concurrent access.

Reply:
Just made that change but it looks like addPeer/delPeer were already acquiring `s.peersLock.Lock()` - unless I am missing something..

Reply:
Yeah you're right, I missed it. Please revert back